brave/brave-core
# Revert "Revert "Use brave search as a default search provider in tor window and update TorWindow NTP""

Reverts brave/brave-core#12390

First commit is reverting previsouly reverted one.
Second commit fixed below issues that has reverted code.
* Fixed default search engines list has two brave entries by hiding from settings UI.
* Fixed onion url is used for searching from normal window by using different keyword.
## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Launch browser and open tor window
2. Search anything from tor window location bar
3. Check onion url is used for searching
4. Go to normal window and set brave as a default search provider
5. Search anything from normal window location bar
6. Check search.brave.com is used for it
7. Re-launch and load brave://settings/searchEngines
8. Check default list has only one brave entry(search.brave.com)
9. Launch tor window and pase`:search.brave4u7jddbv7cyviptqjc7jusxh72uik7zt6adtckl5f4nwy2v72qd.onion` and press tab.
10. Check `:search.brave4u7jddbv7cyviptqjc7jusxh72uik7zt6adtckl5f4nwy2v72qd.onion` works as a provider keyword
